### ISSUE TITLE
Step size works not as expected

### ISSUE THREAD

**Issue by scleronomic** [comet-ml/issue-tracking#355]
```
from comet_ml import Experiment
import tensorflow as tf
from tensorflow.keras.layers import Input, Dense
import numpy as np

project_name = 'test'
exp = Experiment(project_name=project_name,
                 auto_param_logging=False, auto_metric_logging=True)

epochs = 3
n_train = 10
n_test = 4
batch_size = 2
n_dof = 1
x_train = np.random.random((n_train, n_dof))
y_train = np.random.random((n_train, n_dof))
x_test = np.random.random((n_test, n_dof))
y_test = np.random.random((n_test, n_dof))


ip = Input((n_dof,))
x = Dense(units=10)(ip)
x = Dense(units=10)(x)
x = Dense(units=n_dof)(x)
model = tf.keras.models.Model(inputs=ip, outputs=x)

model.compile(optimizer=tf.optimizers.Adam(),
              loss=tf.losses.mean_squared_error)


def print_batch_step(batch, logs):
    print('batch: ', batch)
    print(' step: ', exp.curr_step)


cb = tf.keras.callbacks.LambdaCallback(on_batch_end=print_batch_step)

# A
model.fit(x=x_train, y=y_train, epochs=epochs, batch_size=batch_size,
          validation_data=(x_test, y_test), verbose=0, callbacks=[cb])

# B
exp.set_step(0)
for e in range(epochs):
    model.fit(x=x_train, y=y_train, epochs=e+1, batch_size=batch_size,
              validation_data=(x_test, y_test), initial_epoch=e, verbose=0, callbacks=[cb])

# C
exp.set_step(0)
for e in range(epochs):
    model.fit(x=x_train, y=y_train, epochs=e+1, batch_size=batch_size,
              validation_data=(x_test, y_test), initial_epoch=e, verbose=0, callbacks=[cb])
    exp.set_step((e+1)*(n_train+n_test) // batch_size)
```

* *Output A*:
batch:  1
 step:  1
batch:  2
 step:  1  # -> lags behind 
batch:  3
 step:  1
batch:  4
 step:  1
batch:  0
 step:  7 -> updates to the 5 batches of the first epoch + 2 validation batches
batch:  1
 step:  8 -> updates directly: 
batch:  2
 step:  8 -> lags behind
batch:  3
 step:  8
batch:  4
 step:  8
batch:  0
 step:  14
batch:  1
 step:  15
batch:  2
 step:  15
batch:  3
 step:  15
batch:  4
 step:  15

* Output B:
batch:  0
 step:  0
batch:  1
 step:  1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1
batch:  0
 step:  7  # -> the same as for A 
batch:  1
 step:  1 # -> resets the step back to 1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1
batch:  0
 step:  7
batch:  1
 step:  1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1

* *Output C*:
batch:  0
 step:  0
batch:  1
 step:  1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1
batch:  0
 step:  7 # -> manually set step to (was 7 already in this case)
batch:  1
 step:  1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1
batch:  0
 step:  14 # -> manually set step (was 7 again)
batch:  1
 step:  1
batch:  2
 step:  1
batch:  3
 step:  1
batch:  4
 step:  1

So there are few things here that make me wonder.
* The `step` seems to constantly lag behind
* Why are the validation batches also counted in the steps? Can I change it so that only training batches are counted
* I expected option B to work as well but somehow calling model.fit() seems to reset the step back to zero
* Even if I manually set the step with experiment.set_step(), model.fit() seems to reset the step to zero -> How is it possible to have the correct step when using such epoch loops?

-----
python:        3.7
tensorflow:  2.0.0
comet_ml:   3.1.6

**Comment by dsblank** [comment]
> The step seems to constantly lag behind

The step that is being logged normally comes from keras/tensorflow. If you don't log the step, then it won't change. When you call any of the log methods with a step, step is automatically set inside comet_ml. If you don't log, say, the metric, then step won't change.

> Why are the validation batches also counted in the steps? Can I change it so that only training batches are counted

There are many batches per step. Step doesn't change during the testing of the batches.

> I expected option B to work as well but somehow calling model.fit() seems to reset the step back to zero

Yes, that is what keras/tf does.

> Even if I manually set the step with experiment.set_step(), model.fit() seems to reset the step to zero -> How is it possible to have the correct step when using such epoch loops?

Yes, that is how keras works. If you want to use your own step increments, simply keep track and log those like `experiment.log_metric("loss", value, step=MYSTEP)`

Doe that help?

**Comment by scleronomic** [comment]
Yes this helps. Thank you.
But this means it is basically not possible to use `auto_metric_logging` if you have multiple calls to model.fit(), that's unfortunate.
